dashboard/src/components/Catalog/Catalog.test.tsx

```diff
@@ -435,6 +435,46 @@ describe("pagination and chart fetching", () => {
     spyOnUseState.mockRestore();
     expect(setPage).toHaveBeenCalledWith(2);
   });
+
+  it("resets page (when one of the filters changes", () => {
+    const setState = jest.fn();
+    const setPage = jest.fn();
+    const resetRequestCharts = jest.fn();
+    const charts = {
+      ...defaultChartState,
+      hasFinishedFetching: false,
+      isFetching: false,
+      items: [],
+    } as any;
+    spyOnUseState = jest
+      .spyOn(React, "useState")
+      //  @ts-ignore
+      .mockImplementation((init: any) => {
+        if (init === false) {
+          // Mocking the result of hasLoadedFirstPage to simulate that is already loaded
+          return [true, setState];
+        }
+        if (init === 1) {
+          // Mocking the result of setPage to ensure it's called
+          return [1, setPage];
+        }
+        return [init, setState];
+      });
+
+    const store = getStore({ repos: { repos: [{ metadata: { name: "foo" } } as IAppRepository] } });
+    const wrapper = mountWrapper(
+      store,
+      <Catalog {...populatedProps} resetRequestCharts={resetRequestCharts} charts={charts} />,
+    );
+
+    const input = wrapper.find("input").findWhere(i => i.prop("value") === "foo");
+    input.simulate("change", { target: { value: "foo" } });
+
+    spyOnUseState.mockRestore();
+    expect(setPage).toHaveBeenCalledWith(2); // changes page to 2
+    expect(resetRequestCharts).toHaveBeenCalled(); // but receives a change in a filter
+    expect(wrapper.find("CatalogItems").prop("page")).toBe(1); // and page is again 1
```

Comment:
I believe this is not ensuring what you want. You can verify it commenting the `useEffect(...)` that reset the page and the charts. If you do that, this test will still pass I believe.

I would do a much simpler test, without the `setState` mock. I would just check that after the filter change, `resetRequestCharts` is called twice (I belive it's called once the first time the component is rendered). We would not be checking that the page number is reset but since it's in the same piece of logic it's worth it not to add extra complexity in the test.

Reply:
I just commented 
`  useEffect(() => {
    setPage(1);
    resetRequestCharts();}, ...` and the test is not passing. Besides, when moving to the simpler test, it seems `resetRequestCharts` is called just once (but, I agree with you, it should be twice; I will check it)

Reply:
As per the offline conversation: `setPage` is not really changing the page `expect(wrapper.find("CatalogItems").prop("page")).toBe(2); ---> false`.
Besides, implementing the test for testing this is not that easy. For instance, trying to set a repo search filter with 'checked=true` didn't work.

The agreed decision has been just to add a //TODO here and go ahead with this PR (so that the release can be out asap).